pkg/pipeline/pipeline_project_integration.go

```diff
@@ -0,0 +1,199 @@
+		Project:               unpackProject(d, "project"),
+		FormJSONValues:        formJsonValues,
+	}
+	return projectIntegration, nil
+}
+
+func packProjectIntegration(d *schema.ResourceData, projectIntegration ProjectIntegration) []error {
+	var errors []error
+	setValue := util.MkLens(d)
+
+	errors = setValue("name", projectIntegration.Name)
+	errors = append(errors, setValue("master_integration_id", projectIntegration.MasterIntegrationId)...)
+	errors = append(errors, setValue("master_integration_name", projectIntegration.MasterIntegrationName)...)
+	errors = append(errors, setValue("environments", projectIntegration.Environments)...)
+	errors = append(errors, setValue("is_internal", projectIntegration.IsInternal)...)
+	errors = append(errors, packProject(d, "project", projectIntegration.Project)...)
+
+	return errors
+}
+
+func readProjectIntegration(data *schema.ResourceData, m interface{}) ([]FormJSONValues, error) {
+	log.Printf("[DEBUG] readProjectIntegration")
+	projectIntegration := ProjectIntegration{}
+	resp, err := m.(*resty.Client).R().
+		SetResult(&projectIntegration).
+		Get(projectIntegrationsUrl + "/" + data.Id())
+	log.Println("[DEBUG] projectIntegration body: ", string(json.RawMessage(resp.Body())))
+	if err != nil {
+		return nil, err
+	}
+	log.Println("[DEBUG] projectIntegration Obj: ", projectIntegration)
+	errors := packProjectIntegration(data, projectIntegration)
+	if len(errors) > 0 {
+		return nil, errors[0]
+	}
+	return projectIntegration.FormJSONValues, nil
+}
+
+func createProjectIntegration(data *schema.ResourceData, m interface{}, formValues []FormJSONValues) error {
+	log.Printf("[DEBUG] createProjectIntegration")
+	log.Printf("[TRACE] %+v\n", data)
+
+	projectIntegration, err := unpackProjectIntegration(data, formValues)
+	if err != nil {
+		return err
+	}
+	response := ProjectIntegration{}
+	_, err = m.(*resty.Client).R().
+		SetBody(projectIntegration).
+		SetResult(&response).
+		Post(projectIntegrationsUrl)
+	if err != nil {
+		return err
+	}
+	data.SetId(strconv.Itoa(response.ID))
+
+	return nil
+}
+
+func updateProjectIntegration(data *schema.ResourceData, m interface{}, formValues []FormJSONValues) error {
+	log.Printf("[DEBUG] updateProjectIntegration")
+	log.Printf("[TRACE] %+v\n", data)
+
+	projectIntegration, err := unpackProjectIntegration(data, formValues)
+	if err != nil {
+		return err
+	}
+
+	_, err = m.(*resty.Client).R().SetBody(projectIntegration).Put(projectIntegrationsUrl + "/" + data.Id())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func deleteProjectIntegration(ctx context.Context, data *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] deleteProjectIntegration")
+	log.Printf("[TRACE] %+v\n", data)
+
+	resp, err := m.(*resty.Client).R().
+		Delete(projectIntegrationsUrl + "/" + data.Id())
+
+	if err != nil && resp.StatusCode() == http.StatusNotFound {
+		data.SetId("")
+		return diag.FromErr(err)
```

Comment:
this is contradictory - `404` is expected/ok - so, it's not an error. This should return `nil` and 198 should return `err`